cli/tests/inspector_tests.rs

```diff
@@ -1,62 +1,196 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
 
-use deno_core::futures;
+use deno_core::error::AnyError;
 use deno_core::futures::prelude::*;
 use deno_core::futures::stream::SplitSink;
+use deno_core::futures::stream::SplitStream;
 use deno_core::serde_json;
+use deno_core::serde_json::json;
 use deno_core::url;
 use deno_runtime::deno_fetch::reqwest;
 use deno_runtime::deno_websocket::tokio_tungstenite;
 use deno_runtime::deno_websocket::tokio_tungstenite::tungstenite;
 use std::io::BufRead;
-use std::pin::Pin;
+use std::process::Child;
 use test_util as util;
 use tokio::net::TcpStream;
 use util::http_server;
 
 mod inspector {
   use super::*;
 
-  macro_rules! assert_starts_with {
-  ($string:expr, $($test:expr),+) => {
-    let string = $string; // This might be a function call or something
-    if !($(string.starts_with($test))||+) {
-      panic!("{:?} does not start with {:?}", string, [$($test),+]);
-    }
+  struct InspectorTester {
+    socket_tx: SplitSink<
+      tokio_tungstenite::WebSocketStream<
+        tokio_tungstenite::MaybeTlsStream<TcpStream>,
+      >,
+      tungstenite::Message,
+    >,
+    socket_rx: SplitStream<
+      tokio_tungstenite::WebSocketStream<
+        tokio_tungstenite::MaybeTlsStream<TcpStream>,
+      >,
+    >,
+    notification_filter: Box<dyn FnMut(&str) -> bool + 'static>,
+    child: Child,
+    stderr_lines: Box<dyn Iterator<Item = String>>,
+    stdout_lines: Box<dyn Iterator<Item = String>>,
```

Comment:
Maybe there should be a drop assertion (when not panicking) that ensures these are empty on drop of the tester? Not sure if it's worth it, but it would catch any extra unexpected messages.